DSharpPlus.Caching/Abstractions/AbstractCacheOptions.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+
+namespace DSharpPlus.Caching.Abstractions
+{
+    /// <summary>
+    /// Represents a base option class for all cache options.
+    /// </summary>
+    public abstract class AbstractCacheOptions
+    {
+        /// <summary>
+        /// Specifies a default absolute expiration for all types without a specified expiration time.
+        /// </summary>
+        public TimeSpan DefaultAbsoluteExpiration { get; set; }
+
+        /// <summary>
+        /// Specifies a default sliding expiration for all types without a specified expiration time.
+        /// </summary>
+        public TimeSpan DefaultSlidingExpiration { get; set; }
+
+        /// <summary>
+        /// Stores all absolute expirations for concrete types.
+        /// </summary>
+        public Dictionary<Type, TimeSpan> AbsoluteExpirations { get; set; } = new();
+
+        /// <summary>
+        /// Stores all sliding expirations for concrete types.
+        /// </summary>
+        public Dictionary<Type, TimeSpan> SlidingExpirations { get; set; } = new();
```

Comment:
should be `TimeSpan?`; nullable represents an indefinite lifetime, and negative values cannot be used because the backing store (both `IMemoryCache` and `IDistributedCache`) will throw, and we seem to have a commonality of not wanting "special casings".

Reply:
the issue here is that IMemoryCache doesn't actually support `TimeSpan` everywhere. Take `public static ICacheEntry SetAbsoluteExpiration(this ICacheEntry entry, TimeSpan relative)` - this is the only way to set an absolute expiration on an `IMemoryCache` - or `MemoryCache` for that matter: https://source.dot.net/#Microsoft.Extensions.Caching.Memory/MemoryCache.cs,99099576db12fef5 - and this is true as of [the latest dotnet/runtime commit](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Caching.Abstractions/src/CacheEntryExtensions.cs). nullable TimeSpans are not currently supported by the library.

Reply:
Incorrect; there is an overload for `IMemoryCache` that takes `MemoryCacheOptions`, wherein `AbsoluteExpiration[RelativeToNow]` is nullable; you can also still set the list of post-execution callbacks, so perhaps we could have some default event in the lib (niche, and actually probably bad for perf now that I think about it), but it's a thing. Alternatively check for null, and use a ternary with `Array.Empty<T>` to fill that.

Reply:
Where would that overload be? We can check for null and just not set an expiration, which then leaves responsibility for no expiration to the user (or whoever configures IMemoryCache), though.

Reply:
When I said "check for null" I was referring to the post execution event. 

The overload is on `Set<T>`, which allows you to do all of this (key, value, exp, etc) in one shot.

Reply:
Here's the [MSDN](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.cacheextensions.set?view=dotnet-plat-ext-6.0#microsoft-extensions-caching-memory-cacheextensions-set-1(microsoft-extensions-caching-memory-imemorycache-system-object-0-microsoft-extensions-caching-memory-memorycacheentryoptions))